### ISSUE TITLE
something like initial JVM heap size need for GC heap size in dotnet

### ISSUE THREAD

**Issue by MarshalOfficial** [dotnet/runtime#90481]
I have a REST API application that is highly performance-sensitive, requiring low latency. Upon reviewing the GC metrics, I noticed that the GC operations are quite aggressive. Despite setting a huge value for the GC heap hard limit, the GC is maintaining small heap sizes and performing collections very frequently. In my opinion, it would be beneficial for .NET to include a GC initial heap size option. This option could help prevent GC operations on smaller heap sizes and reduce the performance impact of the GC on REST API response times.

**Comment by mangod9** [comment]
Have you experimented with ConserveMemory setting: https://learn.microsoft.com/en-us/dotnet/core/runtime-config/garbage-collector#conserve-memory to see if you get lower GC pause times?

**Comment by MarshalOfficial** [comment]
> Have you experimented with ConserveMemory setting: https://learn.microsoft.com/en-us/dotnet/core/runtime-config/garbage-collector#conserve-memory to see if you get lower GC pause times?

No, I didn't set value for DOTNET_GCConserveMemory and it would be default value (0).
The only settings that I used in my environment variables:
```
   DOTNET_gcServer: 1
   DOTNET_GCHeapCount: C
   DOTNET_GCHeapHardLimit: 140000000
   DOTNET_GCCpuGroup: 1
```

**Comment by jkotas** [comment]
Setting non-zero `DOTNET_GCConserveMemory` is going to make the GC to try even harder and likely run even more often to conserve memory.

> it would be beneficial for .NET to include a GC initial heap size option

The GC has internal undocumented option to set the smallest gen0 budget: https://github.com/dotnet/runtime/blob/main/src/coreclr/gc/gcconfig.h#L86 . You can experiment with that if you would like to see the GC to run less frequently at the cost of large GC heap size. In general, larger Gen0 and less frequent GCs are going to produce larger latency spikes. When you say that your workload requires low latency, do you mean average latency or 99% latency?

**Comment by MarshalOfficial** [comment]
> Setting non-zero `DOTNET_GCConserveMemory` is going to make the GC to try even harder and likely run even more often to conserve memory.
> 
> > it would be beneficial for .NET to include a GC initial heap size option
> 
> The GC has internal undocumented option to set the smallest gen0 budget: https://github.com/dotnet/runtime/blob/main/src/coreclr/gc/gcconfig.h#L86 . You can experiment with that if you would like to see the GC to run less frequently at the cost of large GC heap size. In general, larger Gen0 and less frequent GCs are going to produce larger latency spikes. When you say that your workload requires low latency, do you mean average latency or 99% latency?

Thanks, I will check Gen0Size option, for latency I mean P(99).
